hopsworksai/internal/structure/cluster.go

```diff
@@ -232,6 +232,26 @@ func flattenAutoscaleConfigurationBase(autoscale *api.AutoscaleConfigurationBase
 	return autoscaleConf
 }
 
+func ExpandAutoscaleConfiguration(autoscaleConfig []interface{}) *api.AutoscaleConfiguration {
+	if autoscaleConfig == nil {
+		return nil
+	}
+	autoscale := &api.AutoscaleConfiguration{}
+	if len(autoscaleConfig) > 0 {
+		autoscaleConfigMap := autoscaleConfig[0].(map[string]interface{})
+		if v, ok := autoscaleConfigMap["non_gpu_workers"]; ok && len(v.([]interface{})) > 0 {
+			config := v.([]interface{})[0].(map[string]interface{})
+			autoscale.NonGPU = ExpandAutoscaleConfigurationBase(config)
+		}
+
+		if v, ok := autoscaleConfigMap["gpu_workers"]; ok && len(v.([]interface{})) > 0 {
+			config := v.([]interface{})[0].(map[string]interface{})
+			autoscale.GPU = ExpandAutoscaleConfigurationBase(config)
+		}
+	}
+	return autoscale
+}
```

Comment:
Why is it here, was it moved from somewhere else?

Reply:
Or is it that we didn't need an expand before?

Reply:
Yes, i was doing it in place in resource_cluster before, but now since i added resource_cluster_from_backup as well i decided to move it here to avoid duplicate code 
https://github.com/logicalclocks/terraform-provider-hopsworksai/pull/17/files#diff-9d4c2a92a936268e0d4c71d0d8d7b75268d96e32241e18e71646cd0090276e92R1061
https://github.com/logicalclocks/terraform-provider-hopsworksai/pull/17/files#diff-01d01655e8656f2ce467f3f3a95c71abc8e608eb5a92f414fcf45c318d11d854R106